source/Halibut.Tests/Timeouts/SendingAndReceivingRequestMessagesTimeoutsFixture.cs

```diff
@@ -10,13 +10,57 @@
 {
     public class SendingAndReceivingRequestMessagesTimeoutsFixture : BaseTest
     {
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testWebSocket: false, testAsyncServicesAsWell: true, testSyncService : false, testSyncClients: false, testAsyncClients: true)]
+        public async Task HalibutTimeoutsAndLimits_ApplyInAsync(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var expectedTimeout = clientAndServiceTestCase.SyncOrAsync == SyncOrAsync.Sync ? HalibutLimits.TcpClientReceiveTimeout : TimeSpan.FromSeconds(10);
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .As<LatestClientAndLatestServiceBuilder>()
+                       .WithPortForwarding(port => PortForwarderUtil.ForwardingToLocalPort(port).WithPortForwarderDataLogging(clientAndServiceTestCase.ServiceConnectionType).Build())
+                       .WithPortForwarding(out var portForwarderRef)
+                       .WithEchoService()
+                       .WithDoSomeActionService(() => portForwarderRef.Value.PauseExistingConnections())
+                       .WhenTestingAsyncClient(clientAndServiceTestCase, b =>
+                       {
+                           b.WithHalibutTimeoutsAndLimits(new HalibutTimeoutsAndLimits()
+                               .SetAllTcpTimeoutsTo(TimeSpan.FromSeconds(133))
+                               .WithTcpClientReceiveTimeout(expectedTimeout));
+                       })
+                       .WithInstantReconnectPollingRetryPolicy()
+                       .Build(CancellationToken))
+            {
+                var echo = clientAndService.CreateClient<IEchoService, IAsyncClientEchoService>();
+                await echo.SayHelloAsync("Make a request to make sure the connection is running, and ready. Lets not measure SSL setup cost.");
+
+                var pauseConnections = clientAndService.CreateClient<IDoSomeActionService, IAsyncClientDoSomeActionService>(IncreasePollingQueueTimeout());
+
+                var sw = Stopwatch.StartNew();
+                var e = (await AssertAsync.Throws<HalibutClientException>(async () => await pauseConnections.ActionAsync())).And;
+                sw.Stop();
+                Logger.Error(e, "Received error");
+                AssertExceptionMessageLooksLikeAReadTimeout(e);
+                sw.Elapsed.Should().BeGreaterThan(expectedTimeout - TimeSpan.FromSeconds(2), "The receive timeout should apply, not the shorter heart beat timeout") // -2s give it a little slack to avoid it timed out slightly too early.
+                    .And
+                    .BeLessThan(expectedTimeout + LowerHalibutLimitsForAllTests.HalfTheTcpReceiveTimeout, "We should be timing out on the tcp receive timeout");
+                
+                // The polling tentacle, will not reconnect in time since it has a 133s receive control message timeout.
```

Comment:
The raises other questions like if the tentacle has sent a large response back, will it be able to move into a control message read/write but timeout on that. Not because the next/proceed took a long time itself but because the data ahead of it resulted in the control messages taking some time.

Reply:
Is this something we need to look at? If so should we card it?